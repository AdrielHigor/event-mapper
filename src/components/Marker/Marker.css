.marker {
  height: 25px;
  width: 25px;
  position: relative;
  top: -33px;
  left: -13px;
}