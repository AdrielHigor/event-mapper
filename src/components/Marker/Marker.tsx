import React from "react";
import "./Marker.css";

export interface IMarker {
  lat: number;
  lng: number;
  onClick: () => void;
}

const Marker = ({ lat, lng, onClick }: IMarker) => (
  <img
    key={`${lat}${lng}`}
    src={require("../../res/images/location-marker.png")}
    alt="Location Marker"
    className="marker"
    onClick={onClick}
  />
);

export default Marker;
