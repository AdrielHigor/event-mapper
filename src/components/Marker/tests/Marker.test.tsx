import renderer from "react-test-renderer";
import Marker, { IMarker } from "../Marker";

jest.mock("../Marker.css", () => ({}));

describe("Marker", () => {
  let component: renderer.ReactTestRenderer;

  const defaultProps: IMarker = {
    lat: 0,
    lng: 1,
    onClick: jest.fn(),
  };

  beforeEach(() => {
    component = renderer.create(<Marker {...defaultProps} />);
  });

  it("should render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should call onClick function when image clicked", () => {
    const img = component.root.findByProps({
      className: "marker",
    });
    img.props.onClick();

    expect(defaultProps.onClick).toHaveBeenCalledTimes(1);
  });
});
