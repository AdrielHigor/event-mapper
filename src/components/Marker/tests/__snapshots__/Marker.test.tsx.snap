// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker should render correctly 1`] = `
<img
  alt="Location Marker"
  className="marker"
  onClick={[MockFunction]}
  src="mockImage"
/>
`;
